src/user.ts

```diff
@@ -0,0 +1,232 @@
+type UserProperties = { [key: string]: unknown }
+
+export class User {
+    public userId: string
+    public orgIdToUserOrgInfo?: OrgIdToUserOrgInfo
+
+    // Metadata about the user
+    public email: string
+    public firstName?: string
+    public lastName?: string
+    public username?: string
+    public properties?: UserProperties
+
+    // If you used our migration APIs to migrate this user from a different system,
+    // this is their original ID from that system.
+    public legacyUserId?: string
+    public impersonatorUserId?: string
+
+    constructor(
```

Comment:
This constructor is going to get harder to use over time, maybe an object here would be better? Or even taking in the other User type?

Reply:
Added a new interface to make it nicer and cleaner